### PR TITLE
Adds in some event props for my next event

### PR DESCRIPTION
**Pull request by Kmc2000**
### Kaboom

This is for an event I have planned, it adds a giant death laser, a battery for it and a planet for it to [redacted]

You can probably guess what it's gonna be but I'd ask that you don't share this too much as some element of surprise is needed.

What's in store for you:

https://drive.google.com/file/d/0B9VUbibptgF2NmVrVDZDYUFUMXM/view?usp=sharing

Expect the event soon (if this is merged)

And yes i've tried to keep the bloat minimal, audio files are at a lower bitrate and the animations are as minimal as possible.

WHY DOES THIS NEED A PR???

"improv" ing  this would be extremely bad, as re-inconning stuff as big as this takes a toll on performance.

If you have any ways to improve the code itself let me know!

###CREDIT TO Keekenox for the sprites for the planet and asteroid belt###

(I will fix the map issue shortly, hang tight)
#### Changelog

:cl:
rscadd: Kmc2000: Added in a giant death laser, event coming soon TM
/:cl:

### CODE REVIEW

**Comment by X-TheDark** [comment]
Quite literally, Mining Asteroid

**Comment by Kmc2000** [comment]
Lol that's one way of putting it

File: code/game/objects/eventprops.dm [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+	var/random
+	var/destroyed = 0
+
+/obj/planet/proc/kaboom()
+	if(!destroyed) //dont blow up an already blown up planet
+		playsound_global('sound/effects/planetdeath.ogg', repeat=0, channel=1, volume=100)
+		random = rand(100, 4000)
+		icon_state = "explosion"
+		name = "debris field"
+		desc = "a husk of a once large planet floating in the distance..it makes you feel remorse for all the people living down there"
```
Comment (ShadowDeath6) [inline code/game/objects/eventprops.dm]: fix caps and punctuation

Comment (Kmc2000) [reply code/game/objects/eventprops.dm]: :+1:

File: code/game/objects/eventprops.dm [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+			shake_camera(M, 15, 1)
+			continue
+		sleep(100)
+		icon_state = "destroyed"
+		priority_announce("Outpost [random]. respond! we just lost contact, are you alright..!? RESPOND!", "Syndicate Open Broadcast", 'sound/AI/commandreport.ogg')
+		sleep(500)
+		random = rand(10000,99999)
+		priority_announce("This is syndicate command, we just lost [random] civilians in a massive strike!", "Syndicate Open Broadcast", 'sound/AI/commandreport.ogg')
+		name = "debris field"
+		desc = "a husk of a once large planet floating in the distance..it makes you feel remorse for all the people that once lived down there"
```
Comment (ShadowDeath6) [inline code/game/objects/eventprops.dm]: dupe

File: code/game/objects/eventprops.dm [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+
+/obj/machinery/firing_console/attack_hand(mob/user) //type that shit to recalibrate!
+	playsound(src,'sound/effects/typing.ogg',25,1)
+	src.say("Recalibration complete")
+
+
+
+
+/obj/machinery/firing_console/attack_ai(mob/user)
+	playsound(src,'sound/effects/typing.ogg',25,1)
```
Comment (ShadowDeath6) [inline code/game/objects/eventprops.dm:138]: is this supposed to do something? as it is it doesn't

Comment (Kmc2000) [reply code/game/objects/eventprops.dm:138]: Fluff device.

Not at all necessary.

File: code/game/objects/railing.dm [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/obj/structure/railing
+	name = "safety railing"
+	desc = "you better not try and climb over this...seriously..."
```
Comment (ShadowDeath6) [inline code/game/objects/railing.dm]: fix punctuation and capitalization

File: code/game/objects/railing.dm [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+	icon_state = "railing"
+	climbable = FALSE//>:^) dont climb over it, it's a fluff thing for my event
```
Comment (ShadowDeath6) [inline code/game/objects/railing.dm]: you might not need this if object/structure is by default not climbable

Comment (Kmc2000) [reply code/game/objects/railing.dm]: OK

File: code/game/turfs/simulated/floor/misc_floor.dm [hunk 6]
```diff
@@ -8,6 +8,53 @@
 	icon_state = "rockvault"
 	floor_tile = /obj/item/stack/tile/plasteel
 
+/turf/open/floor/plating/glass
+	icon_state = "glass9"
+	desc = "A thick duraglass tile, you can see space from here!."
```
Comment (ShadowDeath6) [inline code/game/turfs/simulated/floor/misc_floor.dm]: rogue period

**Comment by Kmc2000** [comment]
Merge ready :tm:

File: code/game/objects/eventprops.dm [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+	icon_state = "fired"
+	sleep(100) //10 seconds
+	icon_state = "poweringdown"
+	sleep(20)
+	icon_state = "base"
+	firing = 0
+	ready = 0
+	for(var/obj/item/device/laserbattery/L in world)
+		L.firing = 0
+		continue
```
Comment (X-TheDark) [inline code/game/objects/eventprops.dm]: Huh? What's the point of this `continue` here?

Comment (Kmc2000) [reply code/game/objects/eventprops.dm]: Is that not needed? thought it made the code carry on after a clause

**Comment by YogsMcree** [comment]
KEEK? THIS IS AN OUTRAGE!,!?,!,!,,!!!
Anyways. When is this planned for. Day of next merge ?

**Comment by Frostben** [comment]
this is some purty stuff

**Comment by Kmc2000** [comment]
@YogsMcree 


Weekend after merge assuming the korean quality checkers keep their schedule

File: code/game/objects/eventprops.dm [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+	can_be_unanchored = 0
+	name = "FLEIJA mkIV laser"
+	desc = "An immensely powerful mounted laser cannon, the immense weight is held with a large truss and mounting points, it is mounted below the station and requires an external power supply to operate"
+	var/ready = 0
+	var/firing = 0
+
+/obj/machinery/deathray/New()
+	START_PROCESSING(SSobj, src)
+
+/obj/machinery/deathray/process()
```
Comment (X-TheDark) [inline code/game/objects/eventprops.dm]: Seeing as the only thing this `process()` does is check for a variable be set, I'd remove all processing stuff altogether... (continued at line 284)

File: code/game/objects/eventprops.dm [hunk 9]
```diff
@@ -0,0 +1,338 @@
...
+		priority_announce("FLEIJA LASER CHARGED: Target: CONFIRMED. PREPARING TO FIRE.", "Incoming Priority Message", 'sound/AI/commandreport.ogg')
+		message_admins("Laser battery at([x],[y],[z] - <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[x];Y=[y];Z=[z]'>JMP</a>) is charged, death ray will fire soon!.")
+		power_drained = 0
+		set_mode(CLAMPED_OFF)
+		warned20 = 0	//reset the warnings
+		warned50 = 0
+		warned70 = 0
+		firing = 1
+		for(var/obj/machinery/deathray/L in world) //fire all lasers in the world
+			L.ready = 1
```
Comment (X-TheDark) [inline code/game/objects/eventprops.dm:273]: ...and instead of setting a var here, make this line `L.charge()`

**Comment by AlpacaGoesCrazy** [comment]
It looks like we are going to blow lavaland

**Comment by ShadowDeath6** [comment]
>destroying lavaland

only in my wildest fantasies

**Comment by Kmc2000** [comment]
ree

**Pull request merged**